src/test/rustdoc/macro-generated-macro.rs

```diff
@@ -1,14 +1,29 @@
-macro_rules! outer {
-    ($($matcher:tt)*) => {
+macro_rules! make_macro {
+    ($macro_name:ident $($matcher:tt)*) => {
         #[macro_export]
-        macro_rules! inner {
+        macro_rules! $macro_name {
             (<= $($matcher)* =>) => {};
         }
     }
 }
 
-// @has macro_generated_macro/macro.inner.html //pre 'macro_rules! inner {'
-// @has - //pre '(<= type $($i : ident) :: * + $e : expr =>) => { ... };'
-outer!(type $($i:ident)::* + $e:expr);
+// @has macro_generated_macro/macro.interpolations.html //pre 'macro_rules! interpolations {'
+// @has - //pre '(<= type $($i:ident)::* + $e:expr =>) => { ... };'
+make_macro!(interpolations type $($i:ident)::* + $e:expr);
+interpolations!(<= type foo::bar + x.sort() =>);
 
-inner!(<= type foo::bar + x.sort() =>);
+// @has macro_generated_macro/macro.attributes.html //pre 'macro_rules! attributes {'
+// @has - //pre '(<= #![no_std] #[cfg(feature = "alloc")] =>) => { ... };'
+make_macro!(attributes #![no_std] #[cfg(feature = "alloc")]);
+
+// @has macro_generated_macro/macro.groups.html //pre 'macro_rules! groups {'
+// @has - //pre '(<= fn {} () { foo[0] } =>) => { ... };'
+make_macro!(groups fn {}() {foo[0]});
+
+// @has macro_generated_macro/macro.linebreak.html //pre 'macro_rules! linebreak {'
+// @has - //pre '    ('
+// @has - //pre '        <= 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25'
+// @has - //pre '        26 27 28 =>'
+// @has - //pre '    ) => { ... };'
+// @has - //pre '};'
```

Comment:
Can you maybe replace this with an `@snapshot` test? It'd look like this:

```rust
// @snapshot macro_linebreak_pre macro_generated_macro/macro.linebreak.html //pre
```

Then you'd run `x test --bless src/test/rustdoc/macro-generated-macro.rs`. It generates a separate file that makes tests less brittle (since you can more easily see diffs). It's also less tedious to write.

Reply:
Yep, sure thing. d1f018b0570952fa2e789b219810e14375846fc7 It puts a bunch of `<span class="...">` in there which would maybe make it harder to review. Not sure if you had that in mind?

Reply:
Ah, good point. It looks like I can fairly easily add support for normalizing the output so it's just text, so I'll open a PR for that. Then the snapshot output looks like this:

```
macro_rules! linebreak {
    (
        <= 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25
        26 27 28 =>
    ) => { ... };
}
```

Reply:
PR: #92914